rust/xaynet-server/src/state_machine/mod.rs

```diff
@@ -329,20 +341,22 @@ impl StateMachineInitializer {
             coordinator_state,
             event_publisher,
             request_rx,
-            self.redis_handle,
-            #[cfg(feature = "model-persistence")]
-            self.s3_handle,
+            self.store,
             #[cfg(feature = "metrics")]
             self.metrics_handle,
         );
 
-        let state_machine = StateMachine::from(PhaseState::<Idle>::new(shared));
+        let state_machine = StateMachine::from(PhaseState::<Idle, _, _>::new(shared));
         (state_machine, request_tx, event_subscriber)
     }
 }
 
 #[cfg(feature = "model-persistence")]
```

Comment:
Do we still need this `#[cfg]` attribute? The store should work even when `mode-persistence` isn't enabled now, since the store internally uses `NoOp`

Reply:
mmh but may `load_global_model` wouldn't work then

Reply:
Right, the problem is that if redis returns a `latest_global_model_id` and the `NoOp` model store returns `None` or an error, the state initializer would fail. 
Maybe it is okay to initialize the state machine without the latest global model, but I want to test this in a separate ticket. Or we can somehow find out at run time which storage implementations are in use and print a warning if the recovery flag is true but the storage doesn't support it.

Reply:
Let's address that properly in another PR then. I think once we have a file backend for models, we should make it the default, and add a `s3-model-persistence` feature instead, that enables the s3 backend.